rest-catalog/rest-catalog-service/src/main/java/org/projectnessie/restcatalog/service/resources/HttpClientHolder.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.restcatalog.service.resources;
+
+import java.net.http.HttpClient;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class HttpClientHolder {
```

Comment:
Best approach would probably be to update Nessie's HTTP client code. I have some approach lying on my machine to refactor the Nessie HTTP client to allow "any" HTTP client, so that non-Quarkus callers would use J11's HTTP client, or HttpURLConnection or another special client (like Vertx or Apache HTTP client or foo bar baz). But that's a bigger change - for later.